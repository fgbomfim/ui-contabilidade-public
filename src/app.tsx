import React from 'react';
import styled from 'styled-components';
import Header from './compositions/header';

export default function App() {
  return (
    <Main>
      <Header />

      <main>Main</main>

      <footer>
        footer
      </footer>
    </Main>
  );
}

const Main = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  min-height: 100vh;
  text-align: center;
`;