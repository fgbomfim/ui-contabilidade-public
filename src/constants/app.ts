
import AccountReceivable from '../assets/image/account-receivable.jpg';
import BankReconciliation from '../assets/image/bank-reconciliation.png';
import CashFlow from '../assets/image/cash-flow.jpg';
import Close from '../assets/image/close-company.jpg';
import DocManager from '../assets/image/doc-manager.jpg';
import Holding from '../assets/image/holding.png';
import Ir from '../assets/image/ir.jpg';
import Legalization from '../assets/image/legalization-company.jpg';
import Mei from '../assets/image/mei.jpg';
import Open from '../assets/image/open-company.jpg';
import TaxConsultancy from '../assets/image/tax-consultancy.jpg';
import TaxPlanning from '../assets/image/tax-planning.webp';
import { Slider } from '../interfaces/slider';

export const SLIDERS: Slider[] = [
  {
    id: 1,
    background: TaxConsultancy,
    text: 'Assessoria Tributária',
    dimension: 0,
  },
  {
    id: 2,
    background: BankReconciliation,
    text: 'Conciliação Bancária',
    dimension: -100,
  },
  {
    id: 3,
    background: TaxPlanning,
    text: 'Planejamento Tributário',
    dimension: -200,
  },
  {
    id: 4,
    background: AccountReceivable,
    text: 'Contas a Receber',
    dimension: -300,
  },
  {
    id: 5,
    background: CashFlow,
    text: 'Fluxo de Caixa',
    dimension: -400,
  },
  {
    id: 6,
    background: DocManager,
    text: 'Gestão de Documentos',
    dimension: -500,
  },
  {
    id: 7,
    background: Ir,
    text: 'Imposto de Renda',
    dimension: -600,
  },
  {
    id: 8,
    background: Legalization,
    text: 'Legalização de Empresas',
    dimension: -700,
  },
  {
    id: 9,
    background: Mei,
    text: 'Atendimento MEI',
    dimension: -800,
  },
  {
    id: 10,
    background: Open,
    text: 'Abertura de Empresas',
    dimension: -900,
  },
  {
    id: 11,
    background: Holding,
    text: 'Estruturção de Negócios',
    dimension: -1000,
  },
  {
    id: 12,
    background: Close,
    text: 'Encerramento de Empresas',
    dimension: -1100,
  },
];