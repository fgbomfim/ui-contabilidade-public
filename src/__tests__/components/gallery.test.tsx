import { render, screen } from '@testing-library/react';

import Gallery from '../../components/gallery';
import { SLIDERS } from '../../constants/app';

jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn(),
}));

describe('Gallery component', () => {
  it('renders without throwing an error', () => {
    const { container } = render(<Gallery />);
    expect(container).toMatchSnapshot();
  });

  it('renders the correct number of sliders', () => {
    render(<Gallery />);
    expect(screen.queryAllByRole('slider')).toHaveLength(SLIDERS.length);
  });
});