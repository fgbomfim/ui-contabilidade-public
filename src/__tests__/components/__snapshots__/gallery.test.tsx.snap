// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component renders without throwing an error 1`] = `
<div>
  <div
    class="sc-blHHSb eDyqre"
  >
    <input
      class="sc-gtLWhw etimfb input1"
      id="s1"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input2"
      id="s2"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input3"
      id="s3"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input4"
      id="s4"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input5"
      id="s5"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input6"
      id="s6"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input7"
      id="s7"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input8"
      id="s8"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input9"
      id="s9"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input10"
      id="s10"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input11"
      id="s11"
      name="slider"
      type="radio"
    />
    <input
      class="sc-gtLWhw etimfb input12"
      id="s12"
      name="slider"
      type="radio"
    />
    <div
      class="sc-egkSDF hryVae slider"
    >
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(tax-consultancy.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService1"
          >
            Assessoria Tributária
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(bank-reconciliation.png);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService2"
          >
            Conciliação Bancária
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(tax-planning.webp);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService3"
          >
            Planejamento Tributário
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(account-receivable.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService4"
          >
            Contas a Receber
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(cash-flow.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService5"
          >
            Fluxo de Caixa
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(doc-manager.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService6"
          >
            Gestão de Documentos
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(ir.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService7"
          >
            Imposto de Renda
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(legalization-company.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService8"
          >
            Legalização de Empresas
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(mei.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService9"
          >
            Atendimento MEI
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(open-company.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService10"
          >
            Abertura de Empresas
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(holding.png);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService11"
          >
            Estruturção de Negócios
          </button>
        </h2>
      </div>
      <div
        class="sc-fAUdSK jAbdjO"
        role="slider"
        style="background-image: url(close-company.jpg);"
      >
        <h2>
          <button
            class="sc-kLhKbu gRbcXn"
            data-testid="goToService12"
          >
            Encerramento de Empresas
          </button>
        </h2>
      </div>
    </div>
    <div
      class="sc-ivxoEo gWqLOz prevNext"
    >
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s12"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s2"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s1"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s3"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s2"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s4"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s3"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s5"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s4"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s6"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s5"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s7"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s6"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s8"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s7"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s9"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s8"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s10"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s9"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s11"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s10"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s12"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
      <div>
        <label
          class="sc-dntaoT dkmgeN"
          for="s11"
        >
          <i
            class="fa-solid fa-arrow-left"
          />
        </label>
        <label
          class="sc-dntaoT dkmgeN"
          for="s1"
        >
          <i
            class="fa-solid fa-arrow-right"
          />
        </label>
      </div>
    </div>
    <div
      class="sc-ghWlax hwrYdA bullets"
    >
      <label
        class="sc-dntaoT dkmgeN"
        for="s1"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s2"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s3"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s4"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s5"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s6"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s7"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s8"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s9"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s10"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s11"
      />
      <label
        class="sc-dntaoT dkmgeN"
        for="s12"
      />
    </div>
  </div>
</div>
`;
